src/main/java/org/altbeacon/beacon/service/scanner/CycledLeScanner.java

```diff
@@ -40,7 +42,10 @@
     private long mScanPeriod;
 
     protected long mBetweenScanPeriod;
-    protected final Handler mHandler = new Handler();
+
+    protected final Handler mHandler = new Handler(Looper.getMainLooper());
+    protected final Handler mScanHandler;
```

Comment:
Is there a reason to use a `HandlerThread` here?  Why not just construct `mScanHandler = new Handler();`?  Then you don't need to clean up the handler thread later.